src/containers/App/App.test.js

```diff
@@ -1,9 +1,71 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
-
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+import React from "react";
+import { shallow } from "enzyme";
+import { App, mapDispatchToProps } from "./App";
+import { getNotes } from "../../thunks";
+
+jest.mock("../../thunks");
+getNotes.mockReturnValue({
+  title: "Hello",
+  noteItems: [{ text: "world", id: 5, isCompleted: "true" }],
+  id: 1
+});
+
+const getNotesMock = jest.fn();
+
+describe("App", () => {
+  let wrapper;
+  window.fetch = jest.fn();
+  beforeEach(() => {
+    wrapper = shallow(<App getNotes={getNotesMock} />);
+  });
+
+  it("should match snapshot and not error", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("should call getNotes on componentDidMount", () => {
+    expect(wrapper.instance().props.getNotes).toHaveBeenCalled();
+  });
+
+  it("should have default state of false for showPopup", () => {
+    expect(wrapper.state("showPopup")).toBe(false);
+  });
+
+  describe("handleClick", () => {
+    it("should change showPopup to true when clicked", () => {
+      //execution
+      wrapper.instance().handleClick();
+
+      //expectation
+      expect(wrapper.state("showPopup")).toBe(true);
+    });
+    it("should change showPopup to false when clicked again", () => {
+      //execution
+      wrapper.instance().handleClick();
+      wrapper.instance().handleClick();
+
+      //expectation
+      expect(wrapper.state("showPopup")).toBe(false);
+    });
+  });
+
+  describe("mapDispatchToProps", () => {
+    let mockDispatch = jest.fn();
+    let props = mapDispatchToProps(mockDispatch);
+
+    it("should call dispatch with correct params", () => {
+      //setup
+      const expected = {
+        title: "Hello",
+        noteItems: [{ text: "world", id: 5, isCompleted: "true" }],
+        id: 1
+      };
+      
+      //execution
+      props.getNotes();
+
+      //expectation
+      expect(mockDispatch).toHaveBeenCalledWith(expected);
```

Comment:
This is difficult to test, thank you for working on this 👍 I like how you mock out the thunk to get the expected value.